### DISCUSSION TITLE
Encountered issues during bacterial segmentation

### DISCUSSION THREAD

**Discussion by zxl555666** [jeanollion/bacmman#16] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/1e70224a-3ae3-4340-b0ff-c1bfc058fdc4)
![image](https://github.com/user-attachments/assets/d5cf4b36-bbd8-441b-892a-a1c759930f11)
![image](https://github.com/user-attachments/assets/bc6338a4-3055-41cf-b1d4-d6e7c861e907)
"Hello everyone, I'd like to inquire about the reason why the bacterial segmentation is not complete and not accurate. As shown in the figure, some bacteria have been segmented, some have not been segmented, and some have been segmented incorrectly. For details on the segmentation situation and parameter settings, please refer to the screenshot. Thank you for your assistance in clarifying this issue."

**Comment by jeanollion** [comment, marked as answer]
Hello  zxl555666, 
I see two reasons: 
1/ your images look blurred, you may have focus issues and/or phase contrast setting issues. This will limit the performance of analysis regardless of which algorithm you use and how you use it. My rule of thumb: if you have doubts on how bacteria should be segmented by eye, no algorithm will perform good enough.  
2/ You are using an old rule-based algorithm that only works on very good quality images. The recommended algorithm for bacterial segmentation and tracking is now [DiSTNet2D](https://github.com/jeanollion/bacmman/wiki/DistNet2D), [paper](https://journals.aps.org/prxlife/abstract/10.1103/PRXLife.2.023004). 
It is a deep learning based algorithm, much more performant and resilient to blurred images. However it has to be trained on data similar to yours, and I have not released yet a training on mother machine data. I will do so in the future but I don't have time right now. 
You can already use DiSTNet ( version 1D)  , starting by using the `ExampleDatasets > dataset1_distnet` configuration from the [online configuration library](https://github.com/jeanollion/bacmman/wiki/Online-Configuration-Library) and tuning it to your data.  

I also notice issues with microchannel segmentation, you should tune the parameter according to their description in the hints. 
Best Regards, 
Jean

**Reply by zxl555666** [reply]
Hello Jean,

Thank you for your detailed response and for taking the time to address my concerns. I appreciate the insight into the potential issues with my images and the algorithm I've been using.

I'm looking forward to trying out DiSTNet (version 1D) and will work on tuning it to my data as you suggested. I'm also aware of the microchannel segmentation issues and will adjust the parameters accordingly.

Thank you for your guidance and support.

Best Regards,
Lei Zhuang
ShanghaiTech University

**Comment by zxl555666** [comment]
![image](https://github.com/user-attachments/assets/34a4c5b8-b1a7-40fc-afd6-bf8992c4fe58)
![image](https://github.com/user-attachments/assets/b67fea7e-4679-4beb-b40b-d9bd3e61530f)
Hello Jean,
I hope this message finds you well. I am currently attempting to use the DiSTNet2D method for segmentation and tracking in my research, but I've encountered an issue while creating a new dataset.
When I try to access the public configurations, I receive the following error message:
Could not get public configurations: Server returned HTTP response code: 403 for URL: https://api.github.com/users/jeanollion/gists?per_page=50
This prevents me from proceeding with the setup process.
I was wondering if you could provide some guidance on how to resolve this issue? I believe it might be related to access permissions or a temporary problem with the GitHub API.
Here are the steps I've taken so far:
1\Confirmed my internet connection is stable and working correctly.
2\Tried refreshing the page and re-attempting the process.
If there's any additional information you need from my side, please let me know.
Thank you for your assistance.
Best regards,
Lei Zhuang
ShanghaiTech University

**Reply by jeanollion** [reply]
Dear Lei, 
Right now I have no problem to access the api from France. 
Can you access this link from your browser ? Could this be an issue related to where you access the link from ? 
Best, 
Jean
